src/layout/mod.rs

```diff
@@ -290,6 +290,20 @@ impl<W: LayoutElement> Layout<W> {
         }
     }
 
+    fn with_options_and_workspaces(config: &Config, options: Options) -> Self {
+        let opts = Rc::new(options);
+
+        let workspaces = (&config.workspaces)
```

Comment:
This should work right?
```suggestion
        let workspaces = config.workspaces
```